search-app/src/cmr/search/api/concepts_lookup.clj

```diff
@@ -0,0 +1,122 @@
+(ns cmr.search.api.concepts-lookup
+  "Defines the API for concepts lookup in the CMR."
+  (:require
+   ;; XXX REMOVE the next require once the service and associations work is complete
+   [cmr-edsc-stubs.core :as stubs]
+   [cmr.common-app.api.routes :as common-routes]
+   [cmr.common.concepts :as concepts]
+   [cmr.common.log :refer (debug info warn error)]
+   [cmr.common.mime-types :as mt]
+   [cmr.common.services.errors :as svc-errors]
+   [cmr.search.api.core :as core-api]
+   [cmr.search.services.query-service :as query-svc]
+   [compojure.core :refer :all]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Constants
+
+(def find-by-concept-id-concept-types
+  #{:collection :granule :variable})
+
+(def supported-concept-id-retrieval-mime-types
+  {:collection #{mt/any
+                 mt/html
+                 mt/xml    ; allows retrieving native format
+                 mt/native ; retrieve in native format
+                 mt/atom
+                 mt/json
+                 mt/echo10
+                 mt/iso19115
+                 mt/iso-smap
+                 mt/dif
+                 mt/dif10
+                 mt/umm-json
+                 mt/legacy-umm-json}
+   :granule #{mt/any
+              mt/xml    ; allows retrieving native format
+              mt/native ; retrieve in native format
+              mt/atom
+              mt/json
+              mt/echo10
+              mt/iso19115
+              mt/iso-smap}
+   :variable #{mt/any
+               mt/xml
+               mt/umm-json}})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Support Functions
+
+(defn- find-concept-by-cmr-concept-id
```

Comment:
I can't believe you didn't refactor this function. 😄

Reply:
heheh